common/perception_utils/include/perception_utils/perception_utils.hpp

```diff
@@ -178,6 +180,25 @@ bool transformObjects(
       tf_target2objects = tf_target2objects_world * tf_objects_world2objects;
       tf2::toMsg(tf_target2objects, object.kinematics.pose_with_covariance.pose);
       // TODO(yukkysaito) transform covariance
+
+      if (transform_shape_footprint) {
+        // convert point from object coordinate(origin is
+        // object.kinematics.pose_with_covariance.pose) to target_frame_id
+        geometry_msgs::msg::TransformStamped tf_target2objects_stamped;
+        tf_target2objects_stamped.transform = tf2::toMsg(tf_target2objects);
+        for (auto & point : object.shape.footprint.points) {
+          // input_msg.header.frame_id to target_frame_id
+          geometry_msgs::msg::Point32 point_transformed;
+          tf2::doTransform<geometry_msgs::msg::Point32>(
+            point, point_transformed, tf_target2objects_stamped);
+
+          // target_frame_id to rotated object coordinate(origin is
+          // object.kinematics.pose_with_covariance.pose)
+          point.x = point_transformed.x - object.kinematics.pose_with_covariance.pose.position.x;
+          point.y = point_transformed.y - object.kinematics.pose_with_covariance.pose.position.y;
+          point.z = point_transformed.z - object.kinematics.pose_with_covariance.pose.position.z;
+        }
+      }
```

Comment:
I have one question :pray: 
Why need this code? The shape footprint is based on object.kinematics.pose_with_covariance.pose as a specification of the message.

Reply:
sorry, already commented by @takayuki5168 
https://github.com/autowarefoundation/autoware.universe/pull/1981#issuecomment-1260935186

Reply:
@Shin-kyoto Can you talk about this later?

Reply:
Removed this from perception_utils and added to detected_object_validation in  https://github.com/autowarefoundation/autoware.universe/pull/1981/commits/64d1c0d6ee4e057db6d6357d53b00a8cff591112

Reply:
@yukkysaito
I used transformPoint and cared the quaternion in https://github.com/autowarefoundation/autoware.universe/commit/35fa236fb5c5c2e99368165d4adf01fabfa16bb7